src/util/time.js

```diff
@@ -0,0 +1,76 @@
+/**
+ * @file time.js
+ */
+
+const findSegmentForTime = (time, timeline) => {
+  const matchingSegments = timeline.filter((x) => {
+    return x.start <= time && time <= x.end;
+  });
+
+  if (matchingSegments.length === 0) {
+    return null;
+  }
+
+  return matchingSegments[0].segment;
+};
+
+const buildMediaTimeline = (playlist) => {
+  const timeline = [];
+
+  if (!playlist.segments || playlist.segments.length === 0) {
+    return;
+  }
+
+  // Assumptions:
+  // - there will always be a segment.duration
+  // - we can start from zero
+  // - segments are in time order
+  // - segment.start and segment.end only come
+  //    from syncController(ignored for now)
+
+  let time = 0;
+
+  playlist.segments.forEach((segment) => {
+    const end = time + segment.duration;
+
+    timeline.push({
+      start: time,
+      end,
+      segment
+    });
+
+    time = end;
+  });
+
+  return timeline;
+};
+
+export const getStreamTime = (player, masterPlaylistController) => {
```

Comment:
Probably should just pass the playlist here instead of MPC.

Reply:
I was not sure if the playlist could potentially change? But I guess it's unlikely that that would happen while calling the method, right?